File: BuscaMedicamentos/src/main/java/IgorDouradoDellEstagio/Medicamento.java
package IgorDouradoDellEstagio;


public class Medicamento implements Comparable<Medicamento>{
    
    
    //Para ocupar menos memória e tornar o programa mais fluído, captei apenas os dados do CSV que variam entre os Medicamentos e podem ser úteis
    private String nome, produto, apresentacao, pisCofins = "";//Instanciei como private para encapsular os objetos
    private boolean em2020 = true;
    private long ean1, registro;
    private double pfSemImp, pmc0; //Por se tratar de uma moeda que pode ter representação decimal importante para a análise, coloquei como tipo "double"
    public Medicamento(String nome, String produto, String apresentacao, long registro, long ean1,
            String pisCofins, double pfSemImp, double pmc0) {//Instancio o construtor
        this.nome = nome;
        this.produto = produto;
        this.apresentacao = apresentacao;
        this.registro = registro;
        this.ean1 = ean1;
        this.pisCofins = pisCofins;
        this.pfSemImp = pfSemImp;
        this.pmc0 = pmc0;
    }

    @Override 
    public int compareTo(Medicamento ordenador) {//Para exercer o comparador entre os Medicamentos baseado no valor de pagamento da bolsa, fiz para ficar na ordem decrescente, dos maiores valores de bolsa até os menores
        if (this.pfSemImp > ordenador.getpfSemImp()) { 
          return -1; 
          } 
          if (this.pfSemImp < ordenador.getpfSemImp()) { 
          return 1; 
          } 
          return 0; 
         }


    //Abaixo os Gets e os Sets para cada atributo do objeto
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public boolean isEm2020() {
        return em2020;
    }

    public void setEm2020(boolean em2020) {
        this.em2020 = em2020;
    }

    public double getPmc0() {
        return pmc0;
    }

    public void setPmc0(double pmc0) {
        this.pmc0 = pmc0;
    }

    public String getproduto() {
        return produto;
    }

    public void setproduto(String produto) {
        this.produto = produto;
    }

    public String getapresentacao() {
        return apresentacao;
    }

    public void setapresentacao(String apresentacao) {
        this.apresentacao = apresentacao;
    }

    public long getregistro() {
        return registro;
    }

    public void setregistro(long registro) {
        this.registro = registro;
    }

    public long getean1() {
        return ean1;
    }
    public void setean1(long ean1) {
        this.ean1 = ean1;
    }
    public String getpisCofins() {
        return pisCofins;
    }
    public void setpisCofins(String pisCofins) {
        this.pisCofins = pisCofins;
    }
    public double getpfSemImp() {
        return pfSemImp;
    }
    public void setpfSemImp(double pfSemImp) {
        this.pfSemImp = pfSemImp;
    }
    
}

File: Python/medicamento.py
class Medicamento :
    # Para ocupar menos mem?ria e tornar o programa mais flu?do, captei apenas os dados do CSV que variam entre os Medicamentos e podem ser ?teis
    nome = None
    produto = None
    apresentacao = None
    pisCofins = ""
    # Instanciei como private para encapsular os objetos
    em2020 = True
    ean1 = 0
    registro = 0
    pfSemImp = 0.0
    pmc0 = 0.0
    # Por se tratar de uma moeda que pode ter representa??o decimal importante para a an?lise, coloquei como tipo "double"
    def __init__(self, nome,  produto,  apresentacao,  registro,  ean1,  pisCofins,  pfSemImp,  pmc0,  em2020) :
        # Instancio o construtor
        self.nome = nome
        self.produto = produto
        self.apresentacao = apresentacao
        self.registro = registro
        self.ean1 = ean1
        self.pisCofins = pisCofins
        self.pfSemImp = pfSemImp
        self.pmc0 = pmc0
        self.em2020 = em2020
    # Abaixo os Gets e os Sets para cada atributo do objeto
    def  getNome(self) :
        return self.nome
    def setNome(self, nome) :
        self.nome = nome
    def  isEm2020(self) :
        return self.em2020
    def setEm2020(self, em2020) :
        self.em2020 = em2020
    def  getPmc0(self) :
        return self.pmc0
    def setPmc0(self, pmc0) :
        self.pmc0 = pmc0
    def  getproduto(self) :
        return self.produto
    def setproduto(self, produto) :
        self.produto = produto
    def  getapresentacao(self) :
        return self.apresentacao
    def setapresentacao(self, apresentacao) :
        self.apresentacao = apresentacao
    def  getregistro(self) :
        return self.registro
    def setregistro(self, registro) :
        self.registro = registro
    def  getean1(self) :
        return self.ean1
    def setean1(self, ean1) :
        self.ean1 = ean1
    def  getpisCofins(self) :
        return self.pisCofins
    def setpisCofins(self, pisCofins) :
        self.pisCofins = pisCofins
    def  getpfSemImp(self) :
        return self.pfSemImp
    def setpfSemImp(self, pfSemImp) :
        self.pfSemImp = pfSemImp